hooks/basic/create_version.py

```diff
@@ -216,9 +217,26 @@ def publish(self, task_settings, item):
         """
 
         sg_publish_data = []
+        pre_processed_paths = []
+
+        # by default let it point to the input path.
+        # if a hook wants to override this, it can use input_path_override
+        input_path = item.properties.path
+
         if "sg_publish_data" in item.properties:
             sg_publish_data.append(item.properties.sg_publish_data)
 
+        if "sg_publish_data_list" in item.properties:
+            sg_publish_data.extend(item.properties.sg_publish_data_list)
```

Comment:
What is this and why is it needed?

Reply:
Ah...found your ticket. Meh, I would just update `sg_publish_data` to be a list rather than have two properties. This feels ham-fisted.

Reply:
I'm just sticking to the SGTK way of naming there properties/variables. Looked at this across other apps too. If an app is dealing with single PublishedFile entities it stores them in sg_publish_data, list of those entities are stored in sg_publish_data_list.

Also, this property will only be populated by the color processing hook, so it also demarkates that.

Reply:
I found no other references to sg_publish_data_list in any sgtk code. Ultimately, I think it is confusing to have both. In which case, I would say lets just deprecate sg_publish_data across the publisher code and use sg_publish_data_list from now on.

Reply:
I was looking at `tank.hook.get_publish_paths(self, sg_publish_data_list)` but that's already deprecated. I'm good with updating all the hooks to use sg_publish_data_list instead. Although, that might create issues for us when we try to integrate more publish hooks like tk-mari, or tk-houdini. And, also when we merge tk-multi-publish2 from shotgunsoftware.

Reply:
We're already so far forked that I don't think this will matter.

Reply:
Also, I don't think Ingest will be the last time we want to publish multiple products from a single Item, so really this is how it should have been from the get go.

Reply:
Alright making the change.